### ISSUE TITLE
[BUG] error: invalid operands to binary expression ('const double' and 'expression')

### ISSUE THREAD

**Issue by yurivict** [bluescarni/heyoka#438]
**Describe the bug**
```
/usr/ports/math/heyoka/work/heyoka-5.1.0/src/model/sgp4.cpp:151:24: error: invalid operands to binary expression ('const double' and 'expression')
    const auto S = MIN(MAX(S0, PERIGE - S1), S1);
                   ~~~~^~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/sys/param.h:332:23: note: expanded from macro 'MAX'
#define MAX(a,b) (((a)>(b))?(a):(b))
                      ^
/usr/include/sys/param.h:331:21: note: expanded from macro 'MIN'
#define MIN(a,b) (((a)<(b))?(a):(b))
                    ^
/usr/ports/math/heyoka/work/heyoka-5.1.0/src/model/sgp4.cpp:151:24: error: invalid operands to binary expression ('const double' and 'expression')
    const auto S = MIN(MAX(S0, PERIGE - S1), S1);
                   ~~~~^~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/sys/param.h:332:23: note: expanded from macro 'MAX'
#define MAX(a,b) (((a)>(b))?(a):(b))
                      ^
/usr/include/sys/param.h:331:30: note: expanded from macro 'MIN'
#define MIN(a,b) (((a)<(b))?(a):(b))
                             ^
2 errors generated.
```

**To Reproduce**
Ewgular build.

**Environment (please complete the following information):**
 - OS: FreeBSD 14.1
 - Installation method: source
 - Version: 5.1.0

clang-18

**Comment by bluescarni** [comment]
@yurivict thanks for the report.

This seems to be an unfortunate name clash with macros in ``/usr/include/sys/param.h``. I have renamed the offending functions in #439, would you mind taking a look?

**Comment by bluescarni** [comment]
Fixed by #439.
